pkgs/sketch_pad/lib/main.dart

```diff
@@ -909,3 +928,22 @@ class _VersionInfoWidgetState extends State<VersionInfoWidget> {
     );
   }
 }
+
+extension GoRouterExtension on BuildContext {
+  void goRetainQueryParams({
+    required String path,
+    required Map<String, String> params,
+    List<String> removeAny = const [],
+  }) {
+    final existingParams = Uri.parse(html.window.location.href).queryParameters;
```

Comment:
Would you instead be able to use [`GoRouterState.of(this)`](https://pub.dev/documentation/go_router/latest/go_router/GoRouterState/of.html) and use the URI exposed from that state?

I haven't used go_router much, so not sure :)

Reply:
It would be great to not have to reach directly to the window's url here. Happy to make changes here (and, John knows go_router well).

Reply:
[Here's the version I came up with](https://github.com/dart-lang/dart-pad/pull/2659/files#diff-3895b9c447708d7e48c8860de55956338bc7bb5104b7589602b6d7c131f129d0). It gets the current URI from the GoRouter's routeInformationProvider instead.